### PR TITLE
Made item’s state text consistent + refactored code

### PR DESCRIPTION
**Pull request by aounhaider1**
fixes https://github.com/eclipse/smarthome/issues/3455
Signed-off-by: Aoun Bukhari <bukhari@itemis.de>

### CODE REVIEW

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.control.js [hunk 1]
```diff
@@ -366,7 +313,7 @@ angular.module('PaperUI.controllers.control', []).controller('ControlPageControl
     $scope.isOptionList = function(item) {
         return (item.stateDescription != null && item.stateDescription.options.length > 0)
     }
-}).controller('ItemController', function($rootScope, $scope, itemService) {
+}).controller('ItemController', function($rootScope, $scope, itemService,util) {
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.control.js]: Shouldn't the formatter add a space between "," and "util"?

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.control.js [hunk 2]
```diff
@@ -161,31 +161,13 @@ angular.module('PaperUI.controllers.control', []).controller('ControlPageControl
                     });
                     item.imageLoaded = false;
                 }
+                item.stateText=util.getItemStateText(item);
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.control.js]: Shouldn't the formatter add spaces in front of and after "="?

**Comment by aounhaider1** [comment]
updated accordingly

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/utility.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                }
+            } else if (!item.stateDescription || !item.stateDescription.pattern) {
+                return state;
+            } else {
+                return sprintf(item.stateDescription.pattern, state);
+            }
+        }
+    }
+});
+function timePrint(pattern, date, dateTime) {
```
Comment (htreu) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/utility.js]: Defining the function here will expose it to the global scope. Please move the function definition into the angular.module.factory() { <here> } block.

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/utility.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                } else {
+                    return $filter('date')(date, "dd.MM.yyyy HH:mm:ss");
+                }
+            } else if (!item.stateDescription || !item.stateDescription.pattern) {
+                return state;
+            } else {
+                return sprintf(item.stateDescription.pattern, state);
+            }
+        }
+    }
```
Comment (htreu) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/utility.js]: Can we add a newline here? Its the last thing, I promise ;-)

Comment (aounhaider1) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/utility.js]: No! :D

Comment (aounhaider1) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/utility.js]: updated :)

**Pull request merged**